sonar-scala-plugin/src/main/scala/org/sonarsource/scala/converter/ScalaConverter.scala

```diff
@@ -164,6 +185,23 @@ class ScalaConverter extends slang.api.ASTConverter {
       new MatchTreeImpl(metaData, convertedExpression, convertedCases.asJava, matchKeyword)
     }
 
+    private def createBinaryExpressionTree(metaData: TreeMetaData, infix: Term.ApplyInfix, operator: BinaryExpressionTree.Operator): slang.api.Tree = {
+      // Scala can have multiple arguments on the right-hand side of an infix function application
+      // Example: foo ** (bar, baz)
+      if (infix.args.length != 1) {
+        return createNativeTree(metaData, infix)
+      }
+      // Scala has special shorthand for an anonymous function
+      // (_ || _) is equivalent to (p1, p2) => p1 || p2
+      val elems = infix.args :+ infix.lhs
+      if (elems.toStream.exists(term => term.is[Term.Placeholder])) {
+        return createNativeTree(metaData, infix)
+      }
+      val leftOperand = convert(infix.lhs)
+      val rightOperand = convert(infix.args.head)
+      val operatorToken = new TokenImpl(textRange(infix.op), infix.op.value, Token.Type.STRING_LITERAL)
```

Comment:
I don't really like that. Ideally, we should re-use the token which is already available from the metadata (we may use something similar to the `keyword` function just below). If we recreate a token, we should at least not declare it as `Token.Type.STRING_LITERAL`.